File: Rally-BackEnd/backend/src/main/java/org/rally/backend/userprofilearm/model/UserInformation.java
package org.rally.backend.userprofilearm.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

@Entity
public class UserInformation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String userName;

    @NotNull
    @Size(min = 3, max = 20)
    private String firstName;

    @NotNull
    @Size(min = 3, max = 20)
    private String lastName;

    @NotNull
    private String neighborhood;

    @NotNull
    private String city;

    @NotNull
    private String state;

    public UserInformation() {
    }

    public UserInformation(String userName, String firstName, String lastName, String neighborhood, String city, String state) {
        this.userName = userName;
        this.firstName = firstName;
        this.lastName = lastName;
        this.neighborhood = neighborhood;
        this.city = city;
        this.state = state;
    }

    public String getState() {
        return state;
    }

    public void setState(String state) {
        this.state = state;
    }

    public String getNeighborhood() {
        return neighborhood;
    }

    public void setNeighborhood(String neighborhood) {
        this.neighborhood = neighborhood;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public int getId() {
        return id;
    }



}

File: Rally-FrontEnd/src/app/user-profile-arm/models/UserInformation.ts
import { ProfilePicture } from "./ProfilePicture";
import { UserEntity } from "./UserEntity";

export class UserInformation {

    id: number;
    user: UserEntity;
    profilePic: ProfilePicture;
    firstName: string;
    lastName: string;
    neighborhood: string;
    city: string;
    state: string;
    
	constructor(id: number, user: UserEntity, firstName: string, lastName: string, neighborhood: string, city: string, state: string){
		this.id = id;
        this.user = user;
		this.firstName = firstName;
        this.lastName = lastName;
        this.neighborhood = neighborhood;
		this.city = city;
        this.state = state;
	}
}